usr.bin/top/top.c

```diff
@@ -907,6 +908,19 @@ main(int argc, const char *argv[])
 				}
 				break;
 
+			    case CMD_grep: /* grep command name */
+				new_message(MT_standout,
+				    "Grep command name: ");
+				if (readline(tempbuf1, sizeof(tempbuf1), false) > 0) {
+					if (tempbuf1[0] == '+' && tempbuf1[1] == '\0') {
+						free(ps.command);
```

Comment:
This free needs to be moved up to before the if since we need to do this in both cases.
Otherwise we leak the previous ps.command

Reply:
Ah, yes, you're right.  It needs to be freed regardless if display is reset or a new value is entered.  Thanks for the quick review, Warner!